### PR TITLE
feat: adds custom prefix support for gettext po

### PR DESCRIPTION
**Pull request by garikkh**
# Description

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

Adds support for custom prefixes in `po-gettext` formatter generated comments. 

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Examples update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [X] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation (if appropriate)

### CODE REVIEW

**Comment by thekip** [comment]
Could you elaborate a bit on the motivation and rationale of this feature?

**Comment by garikkh** [comment]
The main reason is for my TMS - there are times the `-` in the prefix causes issues, either when showing the comments or parsing them into separate lines.

i.e.

```
#. js-lingui:icu=%7BanotherCount%2C+plural%2C+one+%7BSingular+case%7D+other+%7BCase+number+%7BanotherCount%7D%7D%7D&pluralize_on=anotherCount
```

will turn into

```
#. js-
#. lingui:icu=%7BanotherCount%2C+plural%2C+one+%7BSingular+case%7D+other+%7BCase+number+%7BanotherCount%7D%7D%7D&pluralize_on=anotherCount
```

And parsing breaks because it can't find the comment with prefix.

Alternatively I could try to change this to support multi line comments

**Comment by timofei-iatsenko** [comment]
It seems that `#. js-lingui-explicit-id` should also break your TMS, isn't it?

**Comment by vonovak** [comment]
I wonder (based on the given description) if this is something that rather should be fixed on the TMS side, rather than working around their problem in this lib?
I think we need to know why this should be lingui responsibility.

And if this is to be merged, docs need to be added.

**Comment by garikkh** [comment]
> `. js-lingui-explicit-id` should also break your TMS, isn't it?

No, because it breaks in my TMS only because of the line length. PO files are recommended to split on 80 characters, so if your comment is less than that theres no problem. URL encoded ICU strings are likely to be split because they are so long.

I don't know the history/support for the 80 character standard, I think it originates from GNU gettext that does the line splitting as well.

-- as I'm writing this, I just tested and confirmed that gettext _does_ split on 80 characters. 

> I wonder (based on the given description) if this is something that rather should be fixed on the TMS side, rather than working around their problem in this lib?

The above answer is really the only reason why I think it should be supported on lingui - if the standard is 80 character lines, then this isn't a TMS specific issue, it's a PO standard that should be supported (maybe not by doing custom prefix, because that's hacky, but by supporting multi-line icu comments

> And if this is to be merged, docs need to be added.

Yeah, will add docs soon, just wanted to check with the lingui team if this is something you want to merge.

**Comment by garikkh** [comment]
Supporting line split comments is nontrivial as well, for cases like this:

```
#. some other comment
#. js-
#. lingui:pluralize_on=someCount
#. js-
#. lingui-explicit-id
msgid "message_with_id"
msgid_plural "message_with_id_plural"
msgstr[0] "Singular case"
msgstr[1] "Case number {someCount}"
```

**Comment by timofei-iatsenko** [comment]
@garikkh thanks for your investigation. Something new for me. Now, it's more or less clear why your TMS is behaving like that and that is worth to fix. 

I think to avoid this it's better in general to change "-" to "_" underscores, but this will cause a big breaking changes for existing users.

**Comment by garikkh** [comment]
@timofei-iatsenko can I get your review on this? Is there anything else I should add to get this merged?

File: website/docs/ref/catalog-formats.md [hunk 1]
```diff
@@ -144,9 +144,9 @@ export default {
 };
 ```
 
-### Configuration {#po-gettext-configuration}
+### Configuration {#po-gettext-config}
```
Comment (andrii-bodnar) [inline website/docs/ref/catalog-formats.md:147]:
```suggestion
### Configuration {#po-gettext-configuration}
```

File: website/docs/ref/catalog-formats.md [hunk 2]
```diff
@@ -203,6 +210,21 @@ With this format, plural messages are exported in the following ways, depending
 
   Note how `msgid` and `msgid_plural` were extracted from the original message.
 
+- Message **with a custom comment prefix**.
+
+  Some TMS might modify the ICU comment by attempting to split lines to be 80 characters or less,
+  or have trouble reading lingui comments because of the `js-lingui:` prefix. To change the prefix,
+  set `customICUPrefix` to modify the prefix for ICU comments.
```
Comment (andrii-bodnar) [inline website/docs/ref/catalog-formats.md:217]: Please format this as a single line. We don't break long sentences like this. You can use line breaks in your IDE, for example.

**Review by andrii-bodnar — CHANGES_REQUESTED** [review]
@garikkh thanks for the contribution!

File: website/docs/ref/catalog-formats.md (hunk 2 above)
Comment (garikkh) [reply website/docs/ref/catalog-formats.md:217]: Ah, I had autoformatting in my IDE for [this rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/md013.md)

**Review by andrii-bodnar — APPROVED** [review]
@garikkh thank you!

**Pull request merged**